Python/_warnings.c

```diff
@@ -431,6 +431,45 @@ call_show_warning(PyObject *category, PyObject *text, PyObject *message,
     return -1;
 }
 
+static int
+tb_displayline(PyObject *f, PyObject *filename, int lineno, PyObject *name)
+{
+    int err;
+    PyObject *line;
+
+    if (filename == NULL || name == NULL)
+        return -1;
+    line = PyUnicode_FromFormat("  File \"%U\", line %d, in %U\n",
```

Comment:
PyUnicode_FromFormat("%U", arg) doesn't check if arg is really a Unicode type.

I suggest to use %S instead.

Reply:
Sorry I completely forgot I had this PR opened, I'll have to get get back into what I was thinking. 

It's interesting because I just copy and pasted this from traceback.c, so I guess these comments will apply there as well; and I should likely use the `tb_displayline` from traceback.c right ?

Reply:
Right, you must also fix traceback.c.